### ISSUE TITLE
cloudflare_list items linking

### ISSUE THREAD

**Issue by mcatany** [cloudflare/terraform-provider-cloudflare#4469]
### Confirmation

- [X] This is a bug with an existing resource and is not a feature request or enhancement. Feature requests should be submitted with Cloudflare Support or your account team.
- [X] I have searched the issue tracker and my issue isn't already found.
- [X] I have replicated my issue using the latest version of the provider and it is still present.

### Terraform and Cloudflare provider version

```
root@docker-desktop:/# terraform -v
Terraform v1.9.8
on linux_amd64
```

### Affected resource(s)

- cloudflare_list
- cloudflare_list_item

### Terraform configuration files

```hcl
resource "cloudflare_list" "exclude_from_maintenance" {
  account_id  = local.account_id
  name        = "exclude_from_maintenance"
  description = "Domain exclude from maintenance"
  kind        = "hostname"
}

resource "cloudflare_list" "maintenance" {

  account_id  = local.account_id
  name        = "maintenance_x"
  description = "Domain in current maintenance on x"
  kind        = "hostname"
}

resource "cloudflare_list_item" "static_asset" {

  account_id = local.account_id
  list_id    = cloudflare_list.exclude_from_maintenance.id
  hostname {
    url_hostname = "static.x"
  }
}
```

### Link to debug output

https://gist.github.com/mcatany/c10cdc5d8775ce5c03ff0e70f1ec47d1

### Expected output

15:08:44.297 STDOUT terraform: Plan: 0 to add, 0 to change, 0 to destroy.

### Actual output

```
15:08:44.296 STDOUT terraform: Terraform will perform the following actions:
15:08:44.297 STDOUT terraform:   # cloudflare_list.exclude_from_maintenance will be updated in-place
15:08:44.297 STDOUT terraform:   ~ resource "cloudflare_list" "exclude_from_maintenance" {
15:08:44.297 STDOUT terraform:         id          = "baa3671e96b74232847f27cfe3ebfc92"
15:08:44.297 STDOUT terraform:         name        = "exclude_from_maintenance"
15:08:44.297 STDOUT terraform:         # (3 unchanged attributes hidden)
15:08:44.297 STDOUT terraform:       - item {
15:08:44.297 STDOUT terraform:           - value {
15:08:44.297 STDOUT terraform:               - hostname {
15:08:44.297 STDOUT terraform:                   - url_hostname = "static.hivebrite.com" -> null
15:08:44.297 STDOUT terraform:                 }
15:08:44.297 STDOUT terraform:             }
15:08:44.297 STDOUT terraform:         }
15:08:44.297 STDOUT terraform:       - item {
15:08:44.297 STDOUT terraform:           - value {
15:08:44.297 STDOUT terraform:               - hostname {
15:08:44.297 STDOUT terraform:                   - url_hostname = "static.hvbrt.com" -> null
15:08:44.297 STDOUT terraform:                 }
15:08:44.297 STDOUT terraform:             }
15:08:44.297 STDOUT terraform:         }
15:08:44.297 STDOUT terraform:     }
15:08:44.297 STDOUT terraform: Plan: 0 to add, 1 to change, 0 to destroy.

```
### Steps to reproduce

terraform apply

### Additional factoids

Everytime we apply this resources, the list of item shown as to be deleted. If we apply and proceed again, the list of items is shown to be added We tried to deleted all the resources /states/ etc and apply again but the problem is persisting.

**Comment by simonsCatalyst** [comment]
I too have experienced this problem, I suspect this recent change my be suspect: https://github.com/cloudflare/terraform-provider-cloudflare/pull/4228

The problem, at least for me, does not exist in version 4.43.0 but does exist in version 4.44.0 and 4.45.0

**Comment by mbrekhov** [comment]
We have the same problem.
Also solved by downgrading to 4.43.0

**Comment by mcatany** [comment]
any input on this @jacobbednarz?

Thanks
